tests/ui/auxiliary/non-exhaustive-enum.rs

```diff
@@ -0,0 +1,23 @@
+#[non_exhaustive]
+pub enum ErrorKind {
+    NotFound,
+    PermissionDenied,
+    ConnectionRefused,
+    ConnectionReset,
+    ConnectionAborted,
+    NotConnected,
+    AddrInUse,
+    AddrNotAvailable,
+    BrokenPipe,
+    AlreadyExists,
+    WouldBlock,
+    InvalidInput,
+    InvalidData,
+    TimedOut,
+    WriteZero,
+    Interrupted,
+    Other,
+    UnexpectedEof,
+    Unsupported,
+    OutOfMemory,
+}
```

Comment:
Maybe trim this down and/or rename it so it doesn't even seem like we're trying to match std.